tiny_dnn/activations/softmax_layer.h

```diff
@@ -24,8 +24,10 @@ class softmax_layer : public activation_layer {
       y[j] = std::exp(x[j] - alpha);
       denominator += y[j];
     }
+
+    float_t recip_denominator = float_t(1) / denominator;
```

Comment:
When values will not be changed I tend to mark as const. Speed compilation and benefits readability